cypress/stress/polls/polls-in-large-rooms.spec.ts

```diff
@@ -0,0 +1,184 @@
+            if (userNumber >= NUM_USERS) {
+                return;
+            }
+
+            const userName = `user${userNumber}`;
+            cy.getBot(synapse, { displayName: userName, startClient: false }).as(userName).then(
+                user => cy.inviteUser(roomId, user.getUserId()).then(
+                    () => cy.botJoinRoom(user, roomId).then(
+                        () => createUsers(synapse, roomId, userNumber + 1),
+                    ),
+                ),
+            );
+        }
+
+        cy.startSynapse("default").then(
+            _synapse => {
+                synapse = _synapse;
+                cy.initTestUser(synapse, "Jim").then(
+                    () => cy.createRoom({ name: "Large stressful room" }).as("roomId").then(
+                        roomId => createUsers(synapse, roomId, 0),
+                    ),
+                );
+            },
+        );
+    });
+
+    afterEach(() => {
+        cy.stopSynapse(synapse);
+    });
+
+    it("should allow many people to vote in a poll", () => {
+        cy.get<string>("@roomId").then(roomId => {
+            cy.visit('/#/room/' + roomId);
+
+            cy.openMessageComposerOptions().within(() => {
+                cy.get('[aria-label="Poll"]').click();
+            });
+
+            const pollParams = {
+                title: 'Does the polls feature work?',
+                options: ['Yes', 'No', 'Maybe'],
+            };
+            createPoll(pollParams);
+
+            cy.get(".mx_RoomView_body .mx_EventTile").contains(
+                ".mx_EventTile[data-scroll-tokens]",
+                pollParams.title,
+            ).invoke("attr", "data-scroll-tokens").as("pollId");
+
+            cy.get<string>("@pollId").then(pollId => {
+                // All the bots vote for Maybe
+                for (let num = 0; num < NUM_USERS; num++) {
+                    cy.get<MatrixClient>(`@user${num}`).then((user: MatrixClient) =>
+                        botVoteForOption(user, roomId, pollId, pollParams.options[2]),
+                    );
+                }
+
+                // Wait until we can see all the votes have been registered
+                cy.get(
+                    '.mx_MPollBody_totalVotes',
+                    { timeout: NUM_USERS * 200 },
+                ).should('contain', `${NUM_USERS} votes cast`);
+
+                // We vote No
+                getPollOption(pollId, pollParams.options[1]).click('topLeft');
+
+                // The vote counts are visible
+                expectPollOptionVoteCount(pollId, pollParams.options[0], 0);
+                expectPollOptionVoteCount(pollId, pollParams.options[1], 1);
+                expectPollOptionVoteCount(pollId, pollParams.options[2], NUM_USERS);
+            });
+        });
+    });
+});
+
+type CreatePollOptions = {
+    title: string;
+    options: string[];
+};
+
+const createPoll = ({ title, options }: CreatePollOptions) => {
```

Comment:
These could go to poll helpers / commands, so that we can reuse them in other tests.